tools/ci/actions/multicoretests.sh

```diff
@@ -0,0 +1,86 @@
+#!/usr/bin/env bash
+#**************************************************************************
+#*                                                                        *
+#*                                 OCaml                                  *
+#*                                                                        *
+#*                           Samuel Hym, Tarides                          *
+#*                                                                        *
+#*   Copyright 2024 Tarides                                               *
+#*                                                                        *
+#*   All rights reserved.  This file is distributed under the terms of    *
+#*   the GNU Lesser General Public License version 2.1, with the          *
+#*   special exception on linking described in the file LICENSE.          *
+#*                                                                        *
+#**************************************************************************
+
+set -e
+
+OCAMLDIR=ocaml
+DUNEDIR=dune
+MULTICORETESTSDIR=multicoretests
+
+export PREFIX="$HOME/local"
+export PATH="$PREFIX/bin:$PATH"
+
+fatal() {
+  printf %s "$1"
+  exit 1
+}
+
+build_ocaml() {
+  # We let build.yml test for warnings
+  cd "$OCAMLDIR"
+  if ! ./configure --disable-warn-error --disable-stdlib-manpages \
+      --disable-ocamltest --disable-ocamldoc --prefix="$PREFIX" ; then
+    cat config.log
+    exit 1
+  fi
+
+  make -j
+  make install
+}
+
+build_dune() {
+  cd "$DUNEDIR"
+  make release
+  make install PREFIX="$PREFIX"
+}
+
+show_config() {
+  set -x
+  ocamlc -config
+  dune --version
+}
+
+build_testsuite() {
+  cd "$MULTICORETESTSDIR"
+  dune build
+}
+
+run_testsuite() {
+  export QCHECK_MSG_INTERVAL=60
+  cd "$MULTICORETESTSDIR"
+  dune build @ci -j1 --no-buffer --display=quiet --cache=disabled \
```

Comment:
What's the `@ci` target?
Why are you not enabling parallelism?

Reply:
> What's the @ci target?

It's a (re)configurable target that by default will run the test-suite.

> Why are you not enabling parallelism?

Good question - and one we should likely make more clear in the PR text! :pray: 
The test-suite is sensitive to parallelism, to be able to find expected counterexamples, e.g., when misusing a `Hashtbl` in parallel without proper locks. If we schedule too many parallel jobs on top of eachother, the ability to trigger such behaviours degrades. To get the cleanest behaviour we therefore run them one after another. Also, the last time I checked, GitHub action runners only have 3-4 cores, which fits reasonably with the 3 domains of a typical parent with two spawned, parallel children.

Reply:
> What's the `@ci` target?

It’s not predefined by dune, if that’s what you wondered. Is it defined in the root `dune` file:
https://github.com/ocaml-multicore/multicoretests/blob/main/dune#L33